tls/s2n_config.c

```diff
@@ -846,11 +846,11 @@ int s2n_config_enable_cert_req_dss_legacy_compat(struct s2n_config *config)
     return S2N_SUCCESS;
 }
 
-int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb)
+int s2n_config_set_psk_selection_callback(struct s2n_config *config, s2n_psk_selection_callback cb, void *context)
 {
     POSIX_ENSURE_REF(config);
-    POSIX_ENSURE_REF(cb);
```

Comment:
Why is this check no longer needed - why are we relaxing the requirement for cb to be present? Isn't `s2n_psk_selection_callback` necessary here?

Reply:
https://github.com/aws/s2n-tls/pull/2704#discussion_r605991703